zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreter.java

```diff
@@ -227,25 +233,105 @@ public void open() {
 
   @Override
   public void close() {
-    RemoteInterpreterProcess interpreterProcess = getInterpreterProcess();
+    ExecutorService service = Executors.newSingleThreadExecutor();
+    Future future = null;
+    try
+    {
+      future = service.submit(new CloserTask());
+      future.get(10, TimeUnit.SECONDS);
+      return;
+    }
+    catch(InterruptedException | ExecutionException | TimeoutException e)
+    {
+      if(future != null && !future.isDone())
+      {
+        future.cancel(true);
```

Comment:
Do we know if thread interrupting Client.close() will actually kill that thread?

FYI, this is what a thread dump looks like for Client waiting for recv_close event from the other process. Looks like it holds a lock internally. Although in the end the kill/kill 9 will probably make that recv_close error out.
`"Thread-879" #1041 prio=5 os_prio=0 tid=0x00007fc57c012800 nid=0x8a81 runnable [0x00007fc9247cc000]
   java.lang.Thread.State: RUNNABLE
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:170)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	- locked <0x00000000d63de920> (a java.io.BufferedInputStream)
	at org.apache.thrift.transport.TIOStreamTransport.read(TIOStreamTransport.java:127)
	at org.apache.thrift.transport.TTransport.readAll(TTransport.java:86)
	at org.apache.thrift.protocol.TBinaryProtocol.readAll(TBinaryProtocol.java:429)
	at org.apache.thrift.protocol.TBinaryProtocol.readI32(TBinaryProtocol.java:318)
	at org.apache.thrift.protocol.TBinaryProtocol.readMessageBegin(TBinaryProtocol.java:219)
	at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:69)
	at org.apache.zeppelin.interpreter.thrift.RemoteInterpreterService$Client.recv_close(RemoteInterpreterService.java:226)
	at org.apache.zeppelin.interpreter.thrift.RemoteInterpreterService$Client.close(RemoteInterpreterService.java:212)
	at org.apache.zeppelin.interpreter.remote.RemoteInterpreter.close(RemoteInterpreter.java:237)`